discovery: report containerID for containerised services

Yumasi opened the pull request:
<!--
* Contributors are encouraged to read our [CONTRIBUTING](/CONTRIBUTING.md) documentation.
* Both Contributor and Reviewer Checklists are available at https://datadoghq.dev/datadog-agent/guidelines/contributing/#pull-requests.
* The pull request:
  * Should only fix one issue or add one feature at a time.
  * Must update the test suite for the relevant functionality.
  * Should pass all status checks before being reviewed or merged.
* Commit titles should be prefixed with general area of pull request's change.
* Please fill the below sections if possible with relevant information or links.
-->
### What does this PR do?

This PR makes the Discovery agent check add container IDs to discovered services. It relies on the workload metadata store to get the container IDs.

### Motivation

[USMON-1204](https://datadoghq.atlassian.net/browse/USMON-1204?atlOrigin=eyJpIjoiZjUzNDc0MWVjNTc1NGQxY2E0NGUzOTBmMTczOTU2OTAiLCJwIjoiaiJ9)

### Describe how to test/QA your changes

### Possible Drawbacks / Trade-offs

### Additional Notes
<!--
* Anything else we should know when reviewing?
* Include benchmarking information here whenever possible.
* Include info about alternatives that were considered and why the proposed
  version was chosen.
-->

[USMON-1204]: https://datadoghq.atlassian.net/browse/USMON-1204?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

Yumasi committed:
discovery: add containerID field to SD structs

Yumasi committed:
discovery: provide workloadmeta store to discovery check

Yumasi committed:
discovery: add containerID information to discovered services

Yumasi committed:
test: add containerID checks to test

vitkyrka reviewed (commented):
Do you think it would be possible to test this in the E2E test (can be done in a separate PR)?

Yumasi committed:
fix CR

- Only get container ids if there are previous potential services
- Only add container id on confirmed new services
- Add test for propagation of container id values
- Revert unrelated formatting change made by auto-formatter

Yumasi commented on pkg/collector/corechecks/servicediscovery/impl_linux.go:
```diff
@@ -100,6 +110,12 @@ func (li *linuxImpl) DiscoverServices() (*discoveredServices, error) {
 			}
+
+			if id, ok := containers[pid]; ok {
+				svc.service.ContainerID = id
```
Made a new method for handling the `potential->alive` case, and moved the addition of the container id in it.

guyarb commented on pkg/collector/corechecks/servicediscovery/cp_stub.go:
```diff
@@ -0,0 +1,40 @@
+func newContainerProviderStub(targetPIDs []int) proccontainers.ContainerProvider {
+	pidToCid := make(map[int]string)
+
+	for _, pid := range targetPIDs {
+		pidToCid[pid] = "abcd"
+	}
```
what is `abcd`?

If that's only a testutility, then please add `test` build flag

Yumasi replied:
Added the build flag.

> what is abcd?
Just a dummy Container ID used in the test. I moved it to a const value to sync tests and stub code and prevent mistakes.

Yumasi committed:
add test build flag to ContainerProvider stub

Yumasi committed:
test: move dummy container ID to constant

Yumasi commented:
/merge
